File: src/main/java/com/github/kindrat/cassandra/client/ui/DataObject.java
package com.github.kindrat.cassandra.client.ui;

import lombok.*;

import java.util.HashMap;
import java.util.Map;

@ToString
@EqualsAndHashCode
@RequiredArgsConstructor
public class DataObject {
    @Getter
    private final int position;
    private final Map<String, Object> data = new HashMap<>();

    public void set(String name, Object value) {
        data.put(name, value);
    }

    public Object get(String name) {
        if ("#".equals(name)) {
            return position;
        }
        return data.get(name);
    }
}

File: src/main/kotlin/com/github/kindrat/cassandra/client/ui/DataObject.kt
package com.github.kindrat.cassandra.client.ui

import java.util.*

data class DataObject(val position: Int) {
    private val data: MutableMap<String, Any> = HashMap()
    operator fun set(name: String, value: Any) {
        data[name] = value
    }

    operator fun get(name: String): Any {
        return if ("#" == name) {
            position
        } else data[name]!!
    }
}